packages/azure-http-specs/specs/azure/payload/encoding/main.tsp

```diff
@@ -0,0 +1,33 @@
+import "@typespec/http";
+import "@typespec/spector";
+import "@azure-tools/typespec-client-generator-core";
+
+using TypeSpec.Http;
+using global.Azure.ClientGenerator.Core;
+using Spector;
+
+@doc("Test for azure related encode decorator.")
+@scenarioService("/azure/payload/encoding")
+@scenario
+@scenarioDoc("""
+  Test case for azure specific encoding. SDK should generate correct serialization format according to the set encoding.
+  """)
+namespace _Specs_.Azure.Payload.Encoding;
+
+model DurationModel {
+  @encode("duration-constant")
+  input: duration;
+}
+
+@doc("""
+  Test duration with azure specific encoding.
+  Expected request body:
+  ```json
+  {
+    "input": "1.02:59:59.5000000"
+  }
+  ```
+  """)
+@put
+@route("/duration-constant")
+op durationConstant(@body body: DurationModel): NoContentResponse;
```

Comment:
i'm confused: based off of the name, it seems that duration should be a constant, however it isn't specified as one in the spec

Reply:
Actually it isn't though it's name is constant. In swagger we could have "x-ms-format": "duration-constant" which means the serialized payload will be like `1.2:59:59.5000000`. We have discussion with TypeSpec team and the conclusion is that swagger is like this in TypeSpec
```
model Test {
  @encode("duration-constant")
  input: duration;
}
```